### PR TITLE
Comment about a better future where we can get the state diff between two events

### PR DESCRIPTION
**Pull request by MadLittleMods**
Comment about a better future where we can get the state diff between two events

Split off from https://github.com/matrix-org/synapse/pull/13561

Part of https://github.com/matrix-org/synapse/issues/13356

Mentioned in [internal doc](https://docs.google.com/document/d/1lvUoVfYUiy6UaHB6Rb4HicjaJAU40-APue9Q4vzuW3c/edit#bookmark=id.2tvwz3yhcafh)

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [ ] ~~Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)~~
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/handlers/federation_event.py [hunk 1]
```diff
@@ -1031,6 +1031,12 @@ async def _get_state_ids_after_missing_prev_event(
             InvalidResponseError: if the remote homeserver's response contains fields
                 of the wrong type.
         """
+
+        # It would be better if we could query the difference from our known
+        # state to the given `event_id` so the sending server doesn't have to
+        # send as much and we don't have to process as many events. For example
+        # in a room like #matrixhq, we get 200k events (77k state_events, 122k
```
Comment (reivilibre) [inline synapse/handlers/federation_event.py:1038]: either Matrix HQ or #matrix:matrix.org, but not this half-way house :)

```suggestion
        # in a room like #matrix:matrix.org, we get 200k events (77k state_events, 122k
```

**Review by reivilibre — APPROVED** [review]
no objection to adding a comment here, but it feels like it's just going to sit here and never be heard from again.

Is it worth instead having an issue or a spec issue about this? You could then link to it from the comment, even.

File: synapse/handlers/federation_event.py [hunk 2]
```diff
@@ -1031,6 +1031,12 @@ async def _get_state_ids_after_missing_prev_event(
...
+        # auth_events) from this call.
```
Comment (MadLittleMods) [inline synapse/handlers/federation_event.py:1039]: > no objection to adding a comment here, but it feels like it's just going to sit here and never be heard from again.
> 
> Is it worth instead having an issue or a spec issue about this? You could then link to it from the comment, even.
>
> *-- https://github.com/matrix-org/synapse/pull/13586#pullrequestreview-1081874011*

I would like to create a MSC around this.

But perhaps an issue will suffice for now ⏩

Comment (MadLittleMods) [reply synapse/handlers/federation_event.py:1039]: Created https://github.com/matrix-org/synapse/issues/13618

**Comment by MadLittleMods** [comment]
Thanks for the review @reivilibre 🐏

**Pull request merged**